single_tech_samples/synapseanalytics/sample3_integration_testing/README.md

```diff
@@ -0,0 +1,180 @@
+
+#### Software pre-requisites if you don't use dev container<!-- omit in toc -->
+
+- For Windows users, [Windows Subsystem For Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+- [az cli 2.43+](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
+- [az cli - application insights extension](https://docs.microsoft.com/en-us/cli/azure/ext/application-insights/monitor/app-insights?view=azure-cli-latest)
+  - To install, run `az extension add --name application-insights`
+- [Azure DevOps CLI](https://marketplace.visualstudio.com/items?itemName=ms-vsts.cli)
+  - To install, run `az extension add --name azure-devops`
+- [Python 3+](https://www.python.org/)
+- [jq](https://stedolan.github.io/jq/)
+- [makepasswd](https://manpages.debian.org/stretch/makepasswd/makepasswd.1.en.html)
+
+#### Software pre-requisites if you use dev container
+
+- [Docker](https://www.docker.com/)
+- [VSCode](https://code.visualstudio.com/)
+- [Visual Studio Code Remote Development Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+
+  It is strongly recommended to use dev container for the deployment to avoid environment related issues.
+
+### Setup and Deployment
+
+> **IMPORTANT NOTE:** As with all Azure Deployments, this will **incur associated costs**. Remember to teardown all related resources after use to avoid unnecessary costs. See [here](#deployed-resources) for list of deployed resources. See [here](#clean-up) for information on the clean_up script.
+> This deployment was tested using WSL 2 (Ubuntu 20.04) and Debian GNU/Linux 9.9 (stretch)
+
+1. **Initial Setup**
+   - Ensure that:
+      - You have the desired Azure Cloud set. To change from the default (`AzureCloud`), run
+
+        ```bash
+        az cloud set -n <AZURE_CLOUD_NAME>
+        ```
+
+          - For example, if you want to use the Azure US Gov cloud, run
+
+            ```bash
+            az cloud set -n AzureUSGovernment
+            ```
+
+          - To see all of the Azure Cloud options, run
+
+            ```bash
+            az cloud list --output table
+            ```
+
+      - You are logged in to the Azure CLI. To login, run
+
+        ```bash
+        az login
+        ```
+
+      - Azure CLI is targeting the Azure Subscription you want to deploy the resources to. To set target Azure Subscription, run
+
+        ```bash
+        az account set -s <AZURE_SUBSCRIPTION_ID>
+        ```
+
+       Set the following environment variables:
+      - **AZURE_LOCATION** - Azure location to deploy resources. *Default*: `westus`.
+      - **AZURE_SUBSCRIPTION_ID** - Azure subscription id to use to deploy resources. *Default*: default azure subscription. To see your default, run `az account list`.
```

Comment:
worth instructing using that to get a list of available regions they can run `az account list-locations`

Reply:
Sweet, I'm adding a comment for the command `az account list-locations --query [].name --output tsv`